File: scala-hoots/src/main/java/io/hoots/fingerprint/domain/Chunk.java
package io.hoots.fingerprint.domain;

/**
 * Created by rwadowski on 13.04.17.
 */
public class Chunk {

    private final int no;
    private final int size;

    public Chunk(int no, int size) {
        this.no = no;
        this.size = size;
    }

    public int number() {
        return this.no;
    }

    public int getSize() {
        return this.size;
    }

    public float getByte() {
        return this.size * this.no;
    }

    @Override
    public boolean equals(Object object) {
        if(null == object) {
            return false;
        }

        if(this == object) {
            return true;
        }

        if(!(object instanceof Chunk)) {
            return false;
        }

        final Chunk that = (Chunk) object;
        return that.no == this.no;
    }

    @Override
    public int hashCode() {
        int result = 17;
        result = 31 * result + this.no;
        return result;
    }
}

File: scala-hoots/src/main/scala/io/hoots/fingerprint/domain/Chunk.scala
package io.hoots.fingerprint.domain

/**
  * Created by rwadowski on 19.04.17.
  */
case class Chunk(number: ChunkNumber, size: ChunkSize) {

  lazy val chunkByte: Int = number.value * size.value
}

case class ChunkSize(value: Int)

case class ChunkNumber(value: Int) extends Ordered[ChunkNumber] {
  override def compare(that: ChunkNumber): Int = {
    this.value - that.value
  }
}
